### PR TITLE
Fix clippy and tests fail due to simple problems 

### LINKED ISSUE #414: clippy and tests fail due to simple problems
This issue is more of a way to understand the status of this project, as I love it and want to contribute by fixing some bugs. However, I found out that `cargo test` and `clippy` and thus CI don't work due to simple code errors.

Why is everyone ignoring them? Is it intentional or is it okay if someone fixes these errors?

### PR DESCRIPTION
**Pull request by strange-dv**
Closes #414 

This PR fixes small issues that cause CI to fail due to failing tests and `cargo clippy`

## Testing
Local testing using `cargo test --workspace --no-fail-fast --all-features` and `cargo clippy --all --all-targets --all-features`

### CODE REVIEW

File: papergrid/tests/grid/render.rs [hunk 1]
```diff
@@ -210,32 +210,33 @@ test_table!(
...
+// #[ignore = "Fails due to the same reason as the previous test"]
+// test_table!(
+//     doesnt_render_return_carige_1,
+//     grid(2, 2).change_cell((1, 1), "12345678").change_cell((0, 1), "123\r\r\r567").build(),
+//     "+---+--------+"
+//     "|0-0|123\r\r\r567  |"
+//     "+---+--------+"
+//     "|1-0|12345678|"
+//     "+---+--------+"
+// );
```
Comment (strange-dv) [inline papergrid/tests/grid/render.rs:239]: I had to comment out these lines until the problem described in https://github.com/zhiburt/tabled/issues/410 is solved

File: tabled/src/lib.rs [hunk 2]
```diff
@@ -430,21 +430,6 @@ pub use crate::{tabled::Tabled, tables::Table};
...
-///
-/// #[derive(Tabled)]
-/// struct Record {
-///     #[tabled(skip)]
-///     id: [u8; 4],
-///     #[tabled(format("{}.{}.{}.{}.{}", self.id[0], self.id[1], self.id[2], self.id[3], self.name)]
-///     name: String,
-/// }
-/// ```
-///
```
Comment (strange-dv) [inline tabled/src/lib.rs:447]: I couldn't find anything about `#[tabled(format)]` in the docs and assumed that we do not support it anymore so we do not need tests for it. I would be glad to be proven wrong.

File: tabled/src/lib.rs [hunk 3]
```diff
@@ -439,8 +439,8 @@ pub use crate::{tabled::Tabled, tables::Table};
 /// #[derive(Tabled)]
 /// struct Record {
 ///     #[tabled(skip)]
-///     id: [u8; 4],
-///     #[tabled(format("{}.{}.{}.{}.{}", self.id[0], self.id[1], self.id[2], self.id[3], self.name)]
+///     id: u8,
+///     #[tabled(format("{}.{}", self.id, self.name))]
```
Comment (strange-dv) [inline tabled/src/lib.rs:443]: We do not support formatting using complex expressions like `self.id[0]` so I simplified this code a bit

Comment (zhiburt) [reply tabled/src/lib.rs:443]: That's actually one of my endless (not released patches) :smile: 
I did rewrite this support more then a few times.

I wonder what do you think in this regard.
Shall we support functions calls etc?

```rust
#[tabled(format("{}.{}", some_function(self.id), self.name.one_more_field))]
id: u8,
```

Why I did rewrote it is cause it's impossible to support in inline enums :disappointed: 
And having 2 different approaches in 2 different places seeeems like not the right approach.

```rust
#[derive(Tabled)]
enum Something {
#[tabled(inline)]
A {
   #[tabled(format("{}", self.field1, func(0, 1, self)))]  // <- impossible to complex expressions with self 
   field1: usize,
},
B {}
}
```

So anyway maybe you have some thoughts

**Comment by zhiburt** [comment]
Hmmmm

I did not realized it since today.
The errors comes from https://github.com/unicode-rs/unicode-width/issues/65

If on `unicode-width="0.2.11"` it all works but on newest its not.

```log

failures:
    core::table_test::derived::table_emojie_multiline
    settings::render_settings::tab_isnot_handled_by_default_test
failures:
    util::string::tests::strip_test
failures:
    grid::format_configuration::tabs_arent_handled
    grid::render::doesnt_render_return_carige_0
    grid::render::doesnt_render_return_carige_1
    grid::render::hieroglyph_handling_2

```

**Comment by zhiburt** [comment]
I think we need put some thinking to how to resolve it.
Either to force `0.2.11` usage.
Or adjust used logic, accordingly.

File: tabled/src/lib.rs (hunk 3 above)
Comment (strange-dv) [reply tabled/src/lib.rs:443]: What do you mean by rewriting? What do we have now if this is not possible to implement for inline structs? Is this the same functionality I assumed for the test fix? If so, I'm sure that supporting more advanced formatting functionality is a good feature, but obviously, there are many complications that can affect the decision of whether it's worth implementing or not.

If you have patches that extend this functionality beyond the scope of the changed test, what should this test look like to test as much of `#[tabled(format()]` as possible?

**Comment by strange-dv** [comment]
> I think we need put some thinking to how to resolve it.
> Either to force `0.2.11` usage.
> Or adjust used logic, accordingly.

I'm gonna figure out how problematic those adjustments will be and try to fix this issue in this PR since I'm pretty sure enforcing the usage of the outdated version that works for us will cause problems for some users (with `cargo deny` for example).

File: tabled/src/settings/width/wrap.rs [hunk 4]
```diff
@@ -1431,7 +1441,7 @@ mod tests {
                 "\u{1b}[37m(l\u{1b}[39m\u{1b}[37m\u{1b}[41marg\u{1b}[39m\u{1b}[49m\u{1b}[37mest \u{1b}[39m \n",
                 "\u{1b}[37minteger \u{1b}[39m  \n",
                 "\u{1b}[37mless than \u{1b}[39m\n",
-                "\u{1b}[37more equal \u{1b}[39m \n",
+                "\u{1b}[37mor equal \u{1b}[39m \n",
```
Comment (Alexdelia) [inline tabled/src/settings/width/wrap.rs:1444]: that's my fault, sorry guys

Comment (Alexdelia) [reply tabled/src/settings/width/wrap.rs:1444]: I've re-reviewed my previous `fix typos` `MR` to make sure I didn't make other mistakes and I've reverted those error

At the same time, I fix newly found typos

I hope you pardon my mistake and appreciate the `MR`

#416

File: tabled_derive/src/parse/field_attr.rs [hunk 5]
```diff
@@ -15,6 +15,7 @@ pub fn parse_field_attributes(
 }
 
 pub struct FieldAttr {
+    #[allow(dead_code)]
```
Comment (Alexdelia) [inline tabled_derive/src/parse/field_attr.rs:18]: why choose to allow `dead code`?

Comment (strange-dv) [reply tabled_derive/src/parse/field_attr.rs:18]: I'd like to hear from @zhiburt on this.
I see this field was added in https://github.com/zhiburt/tabled/commit/ba048dbc22922ae0584a21d83a6ce10f43dd099e. I know it was a long time ago, but do you remember why it exists but never read or it's fine if I remove it?

As for those `#[allow(dead_code)]` that I left in the test code, I think the reason why they are there is more understandable - they are intended for fields that we never read in tests, but should be able to write to them

Comment (zhiburt) [reply tabled_derive/src/parse/field_attr.rs:18]: > As for those #[allow(dead_code)] that I left in the test code, I think the reason why they are there is more understandable - they are intended for fields that we never read in tests, but should be able to write to them

correct;
though these test could be cleaner I guess

> if I remove it?

Will be all right.

**Comment by zhiburt** [comment]
As far as I think about it nowadays could make a 3 changes in `Cargo.toml`s (`papergrid/Cargo.toml`, `tabled/Cargo.toml`, `testing_table/Cargo.toml`)

```diff
-unicode-width = "0.1"
+unicode-width = "=0.1.11"
```

Once you do the failed tests must be fixed on its own.
And we could merge it.

File: tabled_derive/src/parse/type_attr.rs [hunk 6]
```diff
@@ -14,6 +14,7 @@ pub fn parse_type_attributes(
 }
 
 pub struct TypeAttr {
+    #[allow(dead_code)]
```
Comment (Alexdelia) [inline tabled_derive/src/parse/type_attr.rs:17]: shouldn't we also remove this `dead code`?

Comment (strange-dv) [reply tabled_derive/src/parse/type_attr.rs:17]: Damn, missed it, thanks

Comment (Alexdelia) [reply tabled_derive/src/parse/type_attr.rs:17]: that's what `reviews` are for

**Review by Alexdelia — APPROVED** [review]

**Comment by zhiburt** [comment]
OK seems good to me :100: 

Thanks @Alexdelia for review
Thanks for commitment @strange-dv 

Have a good week;
Take care

**Pull request merged**